influxql/ast.go

```diff
@@ -3636,61 +3656,114 @@ func evalBinaryExpr(expr *BinaryExpr, m map[string]interface{}) interface{} {
 			return lhs != rhs
 		}
 	case float64:
-		rhs, _ := rhs.(float64)
-		switch expr.Op {
-		case EQ:
-			return lhs == rhs
-		case NEQ:
-			return lhs != rhs
-		case LT:
-			return lhs < rhs
-		case LTE:
-			return lhs <= rhs
-		case GT:
-			return lhs > rhs
-		case GTE:
-			return lhs >= rhs
-		case ADD:
-			return lhs + rhs
-		case SUB:
-			return lhs - rhs
-		case MUL:
-			return lhs * rhs
-		case DIV:
-			if rhs == 0 {
-				return float64(0)
+		switch rhs := rhs.(type) {
+		case float64:
+			switch expr.Op {
+			case EQ:
+				return lhs == rhs
+			case NEQ:
+				return lhs != rhs
+			case LT:
+				return lhs < rhs
+			case LTE:
+				return lhs <= rhs
+			case GT:
+				return lhs > rhs
+			case GTE:
+				return lhs >= rhs
+			case ADD:
+				return lhs + rhs
+			case SUB:
+				return lhs - rhs
+			case MUL:
+				return lhs * rhs
+			case DIV:
+				if rhs == 0 {
+					return float64(0)
+				}
+				return lhs / rhs
+			}
+		case int64:
+			switch expr.Op {
+			case EQ:
+				return lhs == float64(rhs)
+			case NEQ:
+				return lhs != float64(rhs)
+			case LT:
+				return lhs < float64(rhs)
+			case LTE:
+				return lhs <= float64(rhs)
+			case GT:
+				return lhs > float64(rhs)
+			case GTE:
+				return lhs >= float64(rhs)
+			case ADD:
+				return lhs + float64(rhs)
+			case SUB:
+				return lhs - float64(rhs)
+			case MUL:
+				return lhs * float64(rhs)
+			case DIV:
+				if rhs == 0 {
+					return float64(0)
+				}
+				return lhs / float64(rhs)
 			}
```

Comment:
Why not type convert `rhs` to `float` and then have a single `switch` on `expr.Op`?